motivaa-backend/motivaa-app/src/main/java/com/motivaa/boundary/validators/ColorValidator.java

```diff
@@ -0,0 +1,16 @@
+package com.motivaa.boundary.validators;
+
+import com.motivaa.control.utility.PossibleColors;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.apache.commons.lang3.EnumUtils;
+
+public class ColorValidator implements ConstraintValidator<ValidColor, String> {
+
+    @Override
+    public boolean isValid(String colorValue, ConstraintValidatorContext context) {
+        if(colorValue == null)
+            return true;
```

Comment:
I know you validate with the NotBlank for this scenario and I guess this is why you return here true, so you don't get validation errors, but I think this is misleadning if someone reads the code. I would merge the two error messages (INVALID_COLOR_RESPONSE and MISSING_COLOR_ERROR_MESSAGE), change the if here to check for string.isBlank() and return false and remove the NotBlank from the validation above.

Reply:
Good recommendation - thanks! Reason I went this way because Swagger didnt mark the color required if I didnt have the NotBlank but there is a work around it (looking back it feels a silly argument!). Will change it!